src/game/client/water/w3dwater.cpp

```diff
@@ -2388,15 +2386,15 @@ void WaterRenderObjClass::Setup_Jba_Water_Shader()
 
         D3DXMATRIX m;
         D3DXMATRIX m2;
-        DX8Wrapper::Get_DX8_Transform(D3DTS_VIEW, (Matrix4 &)m2);
+        DX8Wrapper::Get_DX8_Transform(D3DTS_VIEW, m2);
         float f;
         D3DXMatrixInverse(&m, &f, &m2);
         D3DXMATRIX m3;
         D3DXMatrixScaling(&m3, 0.0625f, 0.0625f, 1.0f);
         D3DXMATRIX m4 = m * m3;
         D3DXMatrixTranslation(&m3, m_riverVOrigin, m_riverVOrigin, 0.0f);
         m4 = m4 * m3;
-        DX8Wrapper::Set_Transform(D3DTS_TEXTURE2, (Matrix4 &)m4);
+        DX8Wrapper::Set_Transform(D3DTS_TEXTURE2, reinterpret_cast<Matrix4 &>(m4));

```

Comment:
This one looks suspicious to me. Should this not call `Set_DX8_Transform` ?

Reply:
According to WB it calls `DX8Wrapper::_Set_DX8_Transform`. So this seems wrong.

Reply:
DX8Wrapper::Set_Transform removed:

![shot_20221002_142609_1](https://user-images.githubusercontent.com/4720891/193454032-988abed0-ed39-4326-9d12-fdaeffac8ca6.jpg)

DX8Wrapper::Set_Transform(D3DTS_TEXTURE2, reinterpret_cast<Matrix4 &>(m4)):

![shot_20221002_142924_1](https://user-images.githubusercontent.com/4720891/193454048-eada3549-c86b-4e94-b55e-720caffac0c9.jpg)

DX8Wrapper::Set_DX8_Transform(D3DTS_TEXTURE2, m4);

![shot_20221002_142734_1](https://user-images.githubusercontent.com/4720891/193454065-9f10fd3b-5871-4b77-a05c-5ae7c188b65b.jpg)

It looks like this code is responsible for the white sparkle on the river water. Both calls Set_Transform and Set_DX8_Transform show the effect. It is difficult to tell the difference. Maybe the effect is flipped horizontally. In any case, looks like `Set_DX8_Transform` is working.

Reply:
Ok. Fixed and pushed.